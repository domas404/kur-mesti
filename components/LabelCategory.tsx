import { Ionicons } from '@expo/vector-icons';
import { View, Text, StyleSheet, Pressable } from 'react-native';
import { Link } from 'expo-router';
import { useState } from 'react';
import { useThemeColor } from '@/hooks/useThemeColor';

type LabelCategory = {
    name: string,
    id: string
}

type LabelCategoryProps = {
    item: LabelCategory,
}

export default function LabelCategory({ item }: LabelCategoryProps) {

    const backgroundColor = useThemeColor({}, 'container');
    const color = useThemeColor({}, 'text');
    const border = useThemeColor({}, 'border');

    return (
        <Link style={[styles.container, {borderColor: border}]} href={`./labeling/${item.id}`} asChild>
            <Pressable style={{ backgroundColor }}>
                {/* <Text style={styles.nameContainer}>{item.name}</Text> */}
                <View style={styles.wasteDisposalContainer}>
                    <View style={styles.headerContainer}>
                        <View style={styles.iconContainer}>
                            <Ionicons name={'cube-outline'} size={24} color={'#E4FFE6'} />
                        </View>
                        <Text style={[styles.wasteDisposalText, {color}]}>{item.name}</Text>
                    </View>
                    <Ionicons name={'arrow-forward'} size={28} color={color} />
                </View>
            </Pressable>
        </Link>
    );
}

const styles = StyleSheet.create({
    container: {
        borderRadius: 20,
        padding: 20,
        width: '100%',
        marginVertical: 4,
        borderWidth: 1,
        // boxShadow: '0 5 12 rgba(0, 0, 0, 0.1)'
    },
    headerContainer: {
        flexDirection: 'row',
        alignItems: 'center',
        gap: 16
    },
    nameContainer: {
        fontSize: 18,
        width: '70%',
        fontWeight: 500
    },
    wasteDisposalContainer: {
        flexDirection: 'row',
        alignItems: 'center',
        paddingVertical: 8,
        width: '100%',
        justifyContent: 'space-between'
    },
    wasteDisposalText: {
        fontSize: 18,
    },
    showMoreButton: {
        flexDirection: 'row',
        alignItems: 'center',
        gap: 4,
        padding: 8,
        alignSelf: 'flex-end'
    },
    showMoreIcon: {
        width: 18
    },
    showMoreContainer: {
        width: '100%',
    },
    iconContainer: {
        backgroundColor: '#86B38A',
		width: 36,
		height: 36,
		borderRadius: 18,
		padding: 6,
    }
});