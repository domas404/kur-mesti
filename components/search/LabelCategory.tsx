import { Ionicons } from '@expo/vector-icons';
import { View, Text, StyleSheet, Pressable } from 'react-native';
import { Link } from 'expo-router';
import { useThemeColor } from '@/hooks/useThemeColor';

type Props = {
    item: {
        name: string,
        id: string
    },
}

export default function LabelCategory({ item }: Props) {

    const { container: backgroundColor, text: color, border } = useThemeColor();
    const { name, id } = item;

    return (
        <Link style={[styles.container, {borderColor: border}]} href={`./labeling/${id}`} asChild>
            <Pressable style={{ backgroundColor }}>
                <View style={styles.wasteDisposalContainer}>
                    <View style={styles.headerContainer}>
                        <View style={styles.iconContainer}>
                            <Ionicons name={'cube-outline'} size={24} color={'#E4FFE6'} />
                        </View>
                        <Text style={[styles.wasteDisposalText, {color}]}>{name}</Text>
                    </View>
                    <Ionicons name={'arrow-forward'} size={28} color={color} />
                </View>
            </Pressable>
        </Link>
    );
}

const styles = StyleSheet.create({
    container: {
        borderRadius: 20,
        padding: 20,
        width: '100%',
        marginVertical: 4,
        borderWidth: 1,
    },
    headerContainer: {
        flexDirection: 'row',
        alignItems: 'center',
        gap: 16
    },
    wasteDisposalContainer: {
        flexDirection: 'row',
        alignItems: 'center',
        paddingVertical: 8,
        width: '100%',
        justifyContent: 'space-between'
    },
    wasteDisposalText: {
        fontSize: 18,
    },
    iconContainer: {
        backgroundColor: '#86B38A',
		width: 36,
		height: 36,
		borderRadius: 18,
		padding: 6,
    }
});