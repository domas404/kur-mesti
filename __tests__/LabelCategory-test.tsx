import { render } from '@testing-library/react-native';

import LabelCategory from '@/components/search/LabelCategory';

jest.mock("expo-font");

describe('<LabelCategory />', () => {
    test('LabelCategory renders correctly', () => {
        const tree = render(<LabelCategory item={{ name: "Plastic", id: "plastic" }} />);
        expect(tree).toMatchSnapshot();
    })
})