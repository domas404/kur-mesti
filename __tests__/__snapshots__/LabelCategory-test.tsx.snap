// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LabelCategory /> LabelCategory renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  href="./labeling/plastic"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  role="link"
  style={
    {
      "backgroundColor": "#fff",
      "borderColor": "#e1e1e1",
      "borderRadius": 20,
      "borderWidth": 1,
      "marginVertical": 4,
      "padding": 20,
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingVertical": 8,
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 16,
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "#86B38A",
            "borderRadius": 18,
            "height": 36,
            "padding": 6,
            "width": 36,
          }
        }
      >
        <Text />
      </View>
      <Text
        style={
          [
            {
              "fontSize": 18,
            },
            {
              "color": "#11181C",
            },
          ]
        }
      >
        Plastic
      </Text>
    </View>
    <Text />
  </View>
</View>
`;
